haystack/backends/solr_backend.py

```diff
@@ -544,23 +544,23 @@ def build_schema(self, fields):
             if field_class.document is True:
                 content_field_name = field_class.index_fieldname
 
-            # DRL_FIXME: Perhaps move to something where, if none of these
-            #            checks succeed, call a custom method on the form that
-            #            returns, per-backend, the right type of storage?
-            if field_class.field_type in ['date', 'datetime']:
-                field_data['type'] = 'date'
-            elif field_class.field_type == 'integer':
-                field_data['type'] = 'long'
-            elif field_class.field_type == 'float':
-                field_data['type'] = 'float'
-            elif field_class.field_type == 'boolean':
-                field_data['type'] = 'boolean'
-            elif field_class.field_type == 'ngram':
-                field_data['type'] = 'ngram'
-            elif field_class.field_type == 'edge_ngram':
-                field_data['type'] = 'edge_ngram'
-            elif field_class.field_type == 'location':
-                field_data['type'] = 'location'
+            try:
+                field_data['type'] = field_class.get_field_type()
+            except AttributeError:
```

Comment:
I like this approach a lot more than what we've been doing but was wondering whether we might want to do something like a `hasattr` check to avoid the possibility of a silent failure if someone's `get_field_type` hit an `AttributeError`.